### PR TITLE
Fixes #8322: When creating a second technique which have the same canonified name but not the same name, ncf builder will silently override the other

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/8322

### CODE REVIEW

File: builder/js/ncf.js [hunk 1]
```diff
@@ -315,6 +333,7 @@ app.controller('ncf-builder', function ($scope, $modal, $http, $log, $location,
       // Select the technique, by using angular.copy to have different objects
       $scope.selectedTechnique=angular.copy(technique);
       $scope.originalTechnique=angular.copy($scope.selectedTechnique);
+      $scope.selectedTechniqueBundleName = $scope.selectedTechnique.bundle_name;
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: Don't duplicate the variable here you don't need to have it :)

File: builder/index.html [hunk 2]
```diff
@@ -186,6 +186,13 @@ <h3 >{{originalTechnique.name}} <span ng-hide="originalTechnique.name">New Techn
                                     Fields over 2048 characters are currently not supported. Please insert your content into a file, and copy it with a file_copy_* method, or use a template.
                                 </span>
                             </div>
+                            <div class="form-group" show-errors>
+                                <label for="bundleName" class="col-sm-3 control-label">Bundle name:</label>
+                                <div class="col-sm-8">
+                                    <input disabled id="bundleName" bundlename name="bundle_name" class="form-control" ng-model="selectedTechniqueBundleName">
```
Comment (VinceMacBuche) [inline builder/index.html]: ng-model should be selectedTechnique.bundle_name

File: builder/js/ncf.js [hunk 3]
```diff
@@ -653,13 +672,24 @@ app.controller('ncf-builder', function ($scope, $modal, $http, $log, $location,
...
 
+  $scope.selectedTechniqueBundleName = "";
+
+  $scope.getBundleName = function (techniqueName) {
+    // Replace all non alpha numeric character (\W is [^a-zA-Z_0-9]) by _
+    return techniqueName ? techniqueName.replace(/\W/g,"_") : "";
+  }
+
+  $scope.updateBundleName = function () {
+    if($scope.selectedTechnique.bundle_name===undefined){
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: Here you should do like you do in the validator and look in original technique !!

Comment (VinceMacBuche) [inline builder/js/ncf.js]: if (originalTechnique === undefined) {
  $scope.selectedTechnique.bundle_name = $scope.getBundleName($scope.selectedTechnique.name);
}

File: builder/js/ncf.js [hunk 4]
```diff
@@ -825,7 +855,6 @@ var cloneModalCtrl = function ($scope, $modalInstance, technique, techniques) {
 };
 
 var SaveChangesModalCtrl = function ($scope, $modalInstance, technique, editForm) {
-  console.log(editForm)
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: Ooops

**Comment by RaphaelGauthier** [comment]
PR updated

**Pull request merged**